### PR TITLE
Make <=> work as expected with strings

### PR DESCRIPTION
**Pull request by watzon**
Currently the spaceship operator with strings just returns the result of running `strcmp`. In reality the spaceship operator is supposed to return -1 if the left side is less than the right side, 1 if the left side is greater than the right side, and 0 if the two sides are equal.

### CODE REVIEW

File: test/natalie/string_test.nat [hunk 1]
```diff
@@ -3,6 +3,20 @@
 require_relative '../spec_helper'
 
 describe 'string' do
+  describe "#<=>" do
+    it "should return -1 if lhs is less than rhs" do
+      ('a' <=> 'b').should == -1
```
Comment (seven1m) [inline test/natalie/string_test.nat]: If you make this `('a' <=> 'z').should == -1`, then the test fails on master.

File: test/natalie/string_test.nat [hunk 2]
```diff
@@ -3,6 +3,20 @@
...
+    end
+
+    it "should return 1 if lhs is greater than rhs" do
+      ('b' <=> 'a').should == 1
```
Comment (seven1m) [inline test/natalie/string_test.nat]: Same here: `('z' <=> 'a')`

**Comment by watzon** [comment]
Changes made :+1: should fail on master now I'd expect

File: lib/natalie/repl.rb [hunk 3]
```diff
@@ -14,33 +14,53 @@ def self.readline(prompt, _)
...
+      compiler.out_path = outfile.path
+      compiler.compile
+      
+      @var_decls = compiler.context[:vars]
+    end
+
+    def completion(str)
+      # TODO
+    end
+
```
Comment (seven1m) [inline lib/natalie/repl.rb]: I think I got a sneak peak of something else you're working on. 😉

Comment (watzon) [reply lib/natalie/repl.rb]: Oops didn't realize the repl stuff got pushed :joy:
Yeah I at least made it not panic on `Ctrl+C` now. I definitely need a little more time with the compiler before I figure out how to make completion work.

**Comment by seven1m** [comment]
Thanks for fixing this!

File: lib/natalie/repl.rb (hunk 3 above)
Comment (seven1m) [reply lib/natalie/repl.rb]: I pulled out the REPL changes and merged this via the terminal. If you think the REPL stuff is good to go, feel free to make a new PR for that! Can't wait!

**Pull request closed without merging**